lib/streamlit/web/server/browser_websocket_handler.py

```diff
@@ -102,7 +102,25 @@ def open(self, *args, **kwargs) -> Optional[Awaitable[None]]:
         else:
             user_info["email"] = email
 
-        self._session_id = self._runtime.create_session(self, user_info)
+        existing_session_id = None
+        try:
+            ws_protocols = [
+                p.strip()
+                for p in self.request.headers["Sec-Websocket-Protocol"].split(",")
+            ]
+
+            if len(ws_protocols) > 1:
+                existing_session_id = ws_protocols[1]
+        except Exception:
+            # Just let existing_session_id=None if we run into any error while trying to
+            # extract it from the Sec-Websocket-Protocol header.
+            pass
```

Comment:
Nit - this feels over-broad. What are the exceptions we could run into here? Is there a particular error we want to discard?

(If we're going to swallow all errors, I think we should at least log them as warnings)

Reply:
Hm, yeah I think the only error we can reasonably expect here is a `KeyError` if `Sec-Websocket-Protocol` isn't set, so let me change this to be more specific